FHEM/14_SD_WS.pm

```diff
@@ -1487,8 +1489,11 @@ sub SD_WS_Parse($$)
 # Wird nur fuer TFA Drop Protokoll benoetigt
 # TFA Drop Protokoll benoetigt als gen 0x31, als key 0xf4
 
-sub SD_WS_LFSR_digest8_reflect($$$$) {
-  my ($bytes, $gen, $key, $rawData) = @_;
+sub SD_WS_LFSR_digest8_reflect {
+  my $bytes = shift // carp 'no bytes data provided';
```

Comment:
Das ganze geht auch mit nur einer Zeile ausreichend Informativ:

`croak qq['Usage: SD_WS_LFSR_digest8_reflect ($bytes, $gen, $key, $rawData)';] if @_ != 4; `

Reply:
Ist das so gewollt, das bei Nutzung von croak bei fehlerhaften Übergaben dann FHEM neu startet???
Egal, wie ich das auch probiere, erfolgt immer ein Neustart.

```
  my $bytes = shift // croak 'Error: called without $hexdata as input';
  my $gen = shift // croak 'Error: called without $hexdata as input';
  my $key = shift // croak 'Error: called without $hexdata as input';
  my $rawData = shift // croak 'Error: called without $hexdata as input';
  # croak 'Usage: SD_WS_LFSR_digest8_reflect' if @_ != 4;
  # croak qq['Usage: SD_WS_LFSR_digest8_reflect ($bytes, $gen, $key, $rawData)';] if @_ != 4;
```

Wenn dem so ist, dann geht das gar nicht!

carp wäre ja noch halbwegs zu ertragen, auch wenn es das Log ziemlich zumüllt.

Wenn wir schon solche (für mich eigentlich sinnlosen Prüfungen) einbauen wollen, dann wäre ein simples return noch das Beste.  Das liefert mir eine PERL WARNING mit der Zeilennummer wo der Aufruf der sub erfolgt und gut ist:

```
2021.09.02 17:58:32 4: sduino_dummy: get rawmsg: MU;P1=247;P2=-750;P3=722;P4=-489;P5=491;P6=-236;P7=-2184;D=1232141456565656145656141456565614141456141414145656141414141456561414141456561414145614561456145614141414141414145614145656145614141732321414565656561456561414565656141414561414141456561414141414565614141414565614141456145614561456141414141414141456141;CP=1;R=55;O;
2021.09.02 17:58:32 4: sduino_dummy: Parse_MU, Fingerprint for MU protocol id 54 -> TFA 30.3233.01 matches, trying to demodulate
2021.09.02 17:58:32 4: sduino_dummy: Parse_MU, last part pair=1 reconstructed, bit=0
2021.09.02 17:58:32 4: sduino_dummy: Parse_MU, Decoded matched MU protocol id 54 dmsg W54#3D9C430618AA01340 length 68 dispatch(1/4) RSSI = -46.5
2021.09.02 17:58:32 4: sduino_dummy: SD_WS_Parse protocol 54, rawData 3D9C430618AA01340
2021.09.02 17:58:32 1: PERL WARNING: Use of uninitialized value $checksum in numeric eq (==) at ./FHEM/14_SD_WS.pm line 467.
2021.09.02 17:58:32 1: PERL WARNING: Use of uninitialized value $checksum in concatenation (.) or string at ./FHEM/14_SD_WS.pm line 470.
2021.09.02 17:58:32 3: sduino_dummy: SD_WS_54 Parse msg W54#3D9C430618AA01340 - ERROR checksum  != 52
2021.09.02 17:58:32 4: sduino_dummy: SD_WS_Parse 3D9C430618AA01340 protocolid 54 (TFA 30.3233.01) - ERROR CRC
```

Reply:
Ja, croak beendet den aktuellen scope.
Das hatte ich nicht bedacht und da das Modul in main läuft, wird eben main beendet.
Croak funktioniert hingegen gut, wenn Module nicht in main laufen.

Für diesen Fall ist carp besser geeignet, da es den laufenden Prozess nicht beendet aber den Aufrufer der Funktion bekannt gibt.